htdocs/expedition/class/expedition.class.php

```diff
@@ -2243,6 +2114,95 @@ public function setClosed()
 		}
 	}
 
+	/**
+	 * Manage Stock MVt onb Close or valid Shipment
+	 * @param      User 	$user        		Object user that modify
+	 * @return     int     <0 if ko, >0 if ok
+	 * @throws Exception
+	 *
+	 */
+	private function manageStockMvtOnEvt($user)
+	{
+		global $langs;
+
+		$error=0;
+
+		require_once DOL_DOCUMENT_ROOT . '/product/stock/class/mouvementstock.class.php';
+
+		$langs->load("agenda");
+
+		// Loop on each product line to add a stock movement
+		$sql = "SELECT cd.fk_product, cd.subprice,";
+		$sql .= " ed.rowid, ed.qty, ed.fk_entrepot,";
+		$sql .= " e.ref,";
+		$sql .= " edb.rowid as edbrowid, edb.eatby, edb.sellby, edb.batch, edb.qty as edbqty, edb.fk_origin_stock";
+		$sql .= " ,cd.rowid as cdid, ed.rowid as edid";
+		$sql .= " FROM " . MAIN_DB_PREFIX . "commandedet as cd,";
+		$sql .= " " . MAIN_DB_PREFIX . "expeditiondet as ed";
+		$sql .= " LEFT JOIN " . MAIN_DB_PREFIX . "expeditiondet_batch as edb on edb.fk_expeditiondet = ed.rowid";
+		$sql .= " INNER JOIN " . MAIN_DB_PREFIX . "expedition as e ON ed.fk_expedition = e.rowid";
+		$sql .= " WHERE ed.fk_expedition = " . ((int) $this->id);
+		$sql .= " AND cd.rowid = ed.fk_origin_line";
+
+		dol_syslog(get_class($this) . "::valid select details", LOG_DEBUG);
+		$resql = $this->db->query($sql);
+		if ($resql) {
+			$cpt = $this->db->num_rows($resql);
+			for ($i = 0; $i < $cpt; $i++) {
+				$obj = $this->db->fetch_object($resql);
+				if (empty($obj->edbrowid)) {
+					$qty = $obj->qty;
+				} else {
+					$qty = $obj->edbqty;
+				}
+				if ($qty <= 0) {
+					continue;
+				}
+				dol_syslog(get_class($this) . "::valid movement index " . $i . " ed.rowid=" . $obj->rowid . " edb.rowid=" . $obj->edbrowid);
+
+				$mouvS = new MouvementStock($this->db);
+				$mouvS->origin = &$this;
+				$mouvS->setOrigin($this->element, $this->id, $obj->cdid, $obj->edid);
```

Comment:
@eldy as you can see here, line $lineIdObjectSrc would be the command line id and $lineIdObjectOrigin would the expedition line id, so I would say origin_line_id for  $lineIdObjectOrigin and maybe origin_src_line_id ?

Reply:
This does not match the setOrigin function parameters.
Inthis function, you have 2 new parameters: 1 to store the id of line and one to store the type of what is the id type (so a string, for example 'order_line', 'shipment_line', ...)

Reply:
The method setOrigin  in MouvementStock class had now this parameters (with this PR). 
That's the goal give more information into Stock mouvement SetOrigin to be more acccurate on why there is a mouvement. The new parameters are defaulted to '', it's harmless for retrocompatibility